src/Classes/NIPSWeb/NIPSWeb_ManagedItem.php

```diff
@@ -71,9 +72,9 @@ protected function __construct($resid, $playlistref = null)
                 );
         $this->folder = $result['folder'];
         $this->title = $result['title'];
-        $this->length = strtotime('1970-01-01 '.$result['length']);
+        $this->length = $result['length'] ?: 0;
```

Comment:
should be no different at all - strtotime just returns the unix timestamp, and I've modified the query to return epoch